plugins/woocommerce/src/Blocks/BlockTypes/AddToCartWithOptions.php

```diff
@@ -129,9 +129,10 @@ protected function render( $attributes, $content, $block ) {
 			* @since 7.6.0
 			* @param boolean.
 			*/
-			$is_disabled_compatibility_layer = apply_filters( 'woocommerce_disable_compatibility_layer', false );
+			$is_disabled_compatibility_layer   = apply_filters( 'woocommerce_disable_compatibility_layer', false );
+			$is_not_purchasable_single_product = ProductType::SIMPLE === $product_type && ( ! $product->is_in_stock() || ! $product->is_purchasable() );
```

Comment:
We have this same check in several files, now. In https://github.com/woocommerce/woocommerce/issues/56558#issuecomment-2785933865 I added a task to evaluate if it would make sense to abstract it into a util.